src/context/fetch.test.ts

```diff
@@ -71,3 +73,57 @@ describe('augmentRequestInit', () => {
     })
   })
 })
+
+describe('createFetchRequestParameters', () => {
+  let mockedRequest: MockedRequest
+
+  beforeAll(() => {
+    mockedRequest = {
+      url: new URL('http://www.mswjs.io'),
+      method: 'GET',
+      headers: new Headers(),
+      cookies: {},
+      mode: 'same-origin',
+      keepalive: true,
+      cache: 'default',
+      destination: '',
+      integrity: '',
+      credentials: 'same-origin',
+      redirect: 'follow',
+      referrer: '',
+      referrerPolicy: 'origin',
+      body: {},
+      bodyUsed: true,
+      params: {},
+    }
+  })
+
+  describe('given a GET request', () => {
+    it('should set the body to null', () => {
+      const result = createFetchRequestParameters(mockedRequest)
+      expect(result.body).toEqual(null)
+    })
+  })
+
+  describe('given a HEAD request', () => {
+    it('should set the body to null', () => {
+      mockedRequest.method = 'HEAD'
+
+      const result = createFetchRequestParameters(mockedRequest)
+      expect(result.body).toEqual(null)
+    })
+  })
+
+  describe('given a POST request', () => {
+    it('should keep the body', () => {
+      const body = {
+        foo: 'bar',
+      }
+      mockedRequest.method = 'POST'
+      mockedRequest.body = body
+
+      const result = createFetchRequestParameters(mockedRequest)
+      expect(result.body).toEqual(JSON.stringify(body))
+    })
+  })
+})
```

Comment:
I think that is better to add an integration test in the file `response-patching.test.ts`

Reply:
Okay I will add an integration test for this. Should I delete the unit tests then? Or should we keep it?

Reply:
I tried my best to add an integration test, but when try to run these tests I always get: `ERROR in ./test/rest-api/response-patching.mocks.ts
    Module not found: Error: Can't resolve 'msw' in '/Users/achim.rolle/workspace/github/msw/test/rest-api'`

Reply:
before running the integration test you should build the library with `yarn build` script.
We can remove the unit for this case. Thanks a lot